dot/sync/errors.go

```diff
@@ -47,6 +47,7 @@ var (
 	errFailedToGetParent            = errors.New("failed to get parent header")
 	errStartAndEndMismatch          = errors.New("request start and end hash are not on the same chain")
 	errFailedToGetDescendant        = errors.New("failed to find descendant block")
+	errBlockOrBodyNil               = errors.New("block or body is nil")
```

Comment:
we should remove that error and the check in `handleBlock`. It's only used with:

```go
		block := &types.Block{
			Header: *bd.Header,
			Body:   *bd.Body,
		}

		if err := s.handleBlock(block); err != nil {
			logger.Debugf("failed to handle block number %d: %s", block.Header.Number, err)
			return err
		}
```

so the block or its body can never be nil. We should nil-panic anyway if we ever pass it a nil block or nil block body in the future with additional code changes.

Reply:
Removed.